src/Schema/Comparator.php

```diff
@@ -429,15 +465,95 @@ public function diffForeignKey(ForeignKeyConstraint $key1, ForeignKeyConstraint
         return $key1->onDelete() !== $key2->onDelete();
     }
 
+    /**
+     * Returns true if there is a difference between the columns
+     */
+    public function compareColumns(Column $column1, Column $column2): bool
+    {
+        if ($this->platform === null) {
+            return count($this->diffColumn($column1, $column2)) !== 0;
+        }
+
+        [$column1, $column2] = $this->stripDefaultValuesFromColumns($column1, $column2);
+
+        if ($this->platform->getColumnSQL('name', $column1) !== $this->platform->getColumnSQL('name', $column2)) {
+            return true;
+        }
+
+        // If the platform supports inline comments, all comparison is already done by the column SQL
+        if ($this->platform->supportsInlineColumnComments()) {
+            return false;
+        }
+
+        // Compare comments
+        if ($column1->getComment() !== $column2->getComment()) {
+            return true;
+        }
+
+        // Compare type comments
+        return $column1->getType() !== $column2->getType()
+            && (
+                $this->platform->isCommentedDoctrineType($column1->getType())
+                || $this->platform->isCommentedDoctrineType($column2->getType())
+            );
+    }
+
+    /**
+     * Remove platform options from both columns if they match the default from
+     * the table or database.
+     *
+     * @return Column[]
+     */
+    private function stripDefaultValuesFromColumns(Column $column1, Column $column2): array
```

Comment:
Stripping properties based on their value may be unnecessarily complex and error-prone. I'd rather use the opposite approach and augment the definition provided by the application with the database defaults.

Reply:
> augment the definition provided by the application with the database defaults

At which point would you do that? The column doesn’t store a reference to the table nor to the database connection.

Reply:
Good question. Regardless of whether we strip the defaults or augment the definition, we'll need the DB connection to understand what the omitted options default to. But at the same time, we want to be able to compare pure definitions, w/o any runtime component.

Even if we keep the approach of stripping the defaults, false positives will be possible. E.g. the offline column has its charset explicitly set to the default value, but the online column has it stripped from the definition. Comparing the two will yield a diff, however, it shouldn't.

Reply:
> But at the same time, we want to be able to compare pure definitions, w/o any runtime component.

It seems to be impossible by definition: whether or not the two definitions are different depends on the runtime. Instead of making the comparator dependent on the platform, it may need to depend on the schema manager which can use both the platform and the connection to normalize the offline column definition.

Reply:
> Even if we keep the approach of stripping the defaults, false positives will be possible. E.g. the offline column has its charset explicitly set to the default value, but the online column has it stripped from the definition. Comparing the two will yield a diff, however, it shouldn't.

That case is already handled by my current implementation. If one or both columns have platform defaults, these defaults are stripped from both columns (online and offline). This way all comparisons should be possible (offline to offline, online to offline and online to online). If two online columns are compared and they have different platform defaults my current implementation does not strip the options.